src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+     *         4) the operator is not already registered
+     */
+    function registerOperator(address operator, bytes32 operatorId, bytes calldata quorumNumbers) external virtual {
+        _registerOperator(operator, operatorId, quorumNumbers);
+    }
+
+    /**
+     * @notice Deregisters the operator with `operatorId` for the specified `quorumNumbers`.
+     * @param operatorId The id of the operator to deregister.
+     * @param quorumNumbers The quourm numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions (these are assumed, not validated in this contract):
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     */
+    function deregisterOperator(bytes32 operatorId, bytes calldata quorumNumbers) external virtual {
+        _deregisterOperator(operatorId, quorumNumbers);
+    }
+
+    /**
+     * @notice Used for updating information on deposits of nodes.
+     * @param operators are the addresses of the operators whose stake information is getting updated
+     * @param operatorIds are the ids of the operators whose stake information is getting updated
+     * @param quorumBitmaps are the bitmap of the quorums that each operator in `operators` is part of
+     * @param prevElements are the elements before this middleware in the operator's linked list within the slasher
+     * @dev Precondition:
+     *          1) `quorumBitmaps[i]` should be the bitmap that represents the quorums that `operators[i]` registered for
+     * @dev reverts if there are no operators registered with index out of bounds
+     */
+    function updateStakes(address[] calldata operators, bytes32[] calldata operatorIds, uint192[] calldata quorumBitmaps, uint256[] calldata prevElements) external {
+        // for each quorum, loop through operators and see if they are apart of the quorum
+        // if they are, get their new weight and update their individual stake history and thes
+        // quorum's total stake history accordingly
+        for (uint8 quorumNumber = 0; quorumNumber < quorumCount;) {
+            OperatorStakeUpdate memory totalStakeUpdate;
+            // for each operator
+            for(uint i = 0; i < operatorIds.length;) {
+                // if the operator is apart of the quorum
+                if (quorumBitmaps[i] >> quorumNumber & 1 == 1) {
+                    // if the total stake has not been loaded yet, load it
+                    if (totalStakeUpdate.updateBlockNumber == 0) {
+                        totalStakeUpdate = _totalStakeHistory[quorumNumber][_totalStakeHistory[quorumNumber].length - 1];
+                    }
```

Comment:
is there a reason for not just doing this before entering this level of the nested loop?

Reply:
in the case where no operators are part of the quorum :woman_shrugging:

Reply:
oh hmm OK, I guess that makes sense.